modules/agar/src/main/ts/ephox/agar/arbitrary/ArbChildrenSchema.ts

```diff
@@ -1,54 +1,75 @@
-import { Arr, Merger, Obj } from '@ephox/katamari';
-import Jsc from '@ephox/wrap-jsverify';
+import { Arr, Fun, Merger, Obj } from '@ephox/katamari';
+import * as fc from 'fast-check';
 
-import { WeightedChoice } from './WeightedChoice';
+import * as WeightedChoice from './WeightedChoice';
 
-/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+type WeightedItem = WeightedChoice.WeightedItem;
+
+interface ChanceItem {
+  readonly chance: number;
+}
+
+interface Detail<T> {
+  readonly components: Record<string, T>;
+}
+
+export interface CompositeDetail extends Detail<WeightedItem> {
+  readonly recursionDepth?: number;
+}
+
+export interface StructureDetail extends Detail<ChanceItem> {}
+
+type Component<T> = T & {
+  readonly component?: string;
+};
+
+export type Construct<T> = (component: string, depth: number) => fc.Arbitrary<T>;
 
 const skipChild = '_';
 
-const toComponents = (detail) =>
+const toComponents = <T>(detail: Detail<T>): Component<T>[] =>
   Obj.mapToArray(detail.components, (v, k) =>
     // If there is no component, then the choice will be None.
     k !== skipChild ? Merger.deepMerge(v, { component: k }) : v
   );
 
-const none = Jsc.constant([]).generator;
+const none = fc.constant([]);
 
-const composite = (rawDepth, detail, construct) => {
+const composite = <T>(rawDepth: number | undefined, detail: CompositeDetail, construct: Construct<T>): fc.Arbitrary<T[]> => {
```

Comment:
This function is O(n<sup>2</sup>) and is called recursively which was fine for jsverify apparently as it in most cases picked array lengths that were <5, however fast-check picked much larger values and having 5<sup>2</sup> child elements being created was about the limit before the test started taking too long.